client/my-sites/controller.js

```diff
@@ -538,6 +539,13 @@ export function siteSelection( context, next ) {
 
 	const shouldRenderNoSites = ! context.section.enableNoSites && ! isUnlinkedCheckout;
 
+	// Immediately continue if the user is logged out and coming from My Jetpack
+	// so it gets picked up by the checkout middleware and redirects to logged-out checkout
+	if ( isContextSourceMyJetpack( context ) && ! currentUser ) {
```

Comment:
If the new middleware function for My Jetpack redirects to the siteless checkout paths, could we get away without this condition in the `siteSelection` middleware function?